rebasehelper/cli.py

```diff
@@ -116,163 +114,26 @@ class CLI(object):
-            choices=['always', 'never', 'auto'],
-            default='auto',
-            help="colorize the output, defaults to %(default)s"
-        )
-        parser.add_argument(
-            "--disable-inapplicable-patches",
-            default=False,
-            action="store_true",
-            dest='disable_inapplicable_patches',
-            help="disable inapplicable patches in rebased SPEC file"
-        )
-        parser.add_argument(
-            "--non-interactive",
-            default=False,
-            action="store_true",
-            dest='non_interactive',
-            help="do not interact with user"
-        )
-        parser.add_argument(
-            "--build-tasks",
-            dest="build_tasks",
-            metavar="OLD_TASK,NEW_TASK",
-            type=lambda s: s.split(','),
-            help="comma-separated remote build task ids"
-        )
-        parser.add_argument(
-            "--builds-nowait",
-            default=False,
-            action="store_true",
-            help="do not wait for remote builds to finish"
-        )
-        parser.add_argument(
-            "--builder-options",
-            default=None,
-            metavar="BUILDER_OPTIONS",
-            help="enable arbitrary local builder option(s), enclose %(metavar)s in quotes "
-                 "and note that = before it is mandatory"
-        )
-        parser.add_argument(
-            "--srpm-builder-options",
-            default=None,
-            metavar="SRPM_BUILDER_OPTIONS",
-            help="enable arbitrary local srpm builder option(s), enclose %(metavar)s in quotes "
-                 "and note that = before it is mandatory"
-        )
-        parser.add_argument(
-            "--results-dir",
-            help="directory where rebase-helper output will be stored"
-        )
-        parser.add_argument(
-            "--get-old-build-from-koji",
-            default=False,
-            action="store_true",
-            help="do not build old sources, download latest build from Koji instead"
-        )
-        parser.add_argument(
-            "sources",
-            metavar='SOURCES',
-            nargs='?',
-            default=None,
-            help="new upstream sources"
-        )
-        parser.add_argument(
-            "--changelog-entry",
-            default="- New upstream release %{version}",
-            help="text to use as changelog entry, can contain RPM macros, which will be expanded"
-        )
+        previous_group = 0
+        for option in traverse(OPTIONS):
+            option_kwargs = dict(option)
+            for key in ("group", "default", "name"):
+                if key in option_kwargs:
+                    del option_kwargs[key]
+
+            if "group" in option:
+                if previous_group != option["group"]:
+                    group = parser.add_mutually_exclusive_group()
+                actions_container = group
+                previous_group = option["group"]
```

Comment:
This could be moved two lines up, it is more logical that way IMHO.

Reply:
Also the variable could be named `current_group`, it makes more sense since you are counting the groups from 0.